src/appleseedmaya/exporters/shadingnodeexporter.cpp

```diff
@@ -58,6 +58,12 @@ namespace asr = renderer;
 namespace
 {
 
+const char *g_compToColorParamNames[] = {"compR", "compG", "compB"};
+const char *g_compToColorOutputParamName = "comp";
+
+const char *g_colorToCompParamNames[] = {"compR", "compG", "compB"};
```

Comment:
It's no coincidence that several constants and arrays have exactly identical values.

That's why I would suggest removing only strictly redundant constants and arrays and renaming them appropriately.

Basically you need:

* One array for `"compR", "compG", "compB"`
* One array for `"compX", "compY", "compZ"`
* One array for `"compU", "compV"`
* One constant for `"comp"`